File: 3340-check-balanced-string.js
// ❓ DESCRIPTION:
// You are given a string num consisting of only digits. A string of digits is called balanced if the sum of the digits at even indices is equal to the sum of digits at odd indices.
// Return true if num is balanced, otherwise return false.

// ✅ SOLUTION (100% runtime, 87.14% memory):
/**
 * @param {string} num
 * @return {boolean}
 */
const isBalanced = function(num) {
    let sumOfEvens = 0;
    let sumOfOdds = 0;
    for (let i = 0; i < num.length; i++) {
        if (i % 2 === 0) {
            sumOfEvens += +num[i];
        } else {
            sumOfOdds += +num[i];
        }
    }
    return sumOfOdds === sumOfEvens;
};

// 📌 TESTCASE:
console.log(isBalanced("1234")) // false
console.log(isBalanced("24123")) // true

File: 3340-check-balanced-string.py
class Solution(object):
    def isBalanced(self, num):
        sum_evens = 0
        sum_odds = 0
        for i in range(len(num)):
            if i % 2 == 0:
                sum_evens += int(num[i])
            else:
                sum_odds += int(num[i])
        return sum_evens == sum_odds
    
s = Solution()
print(s.isBalanced('1234'))
print(s.isBalanced('24123'))
        